includes/class-wc-payment-gateway-wcpay.php

```diff
@@ -754,7 +755,11 @@ public function process_payment( $order_id ) {
 				];
 			}
 
-			UPE_Payment_Gateway::remove_upe_payment_intent_from_session();
+			if ( WC_Payments_Features::is_upe_split_enabled() || WC_Payments_Features::is_upe_deferred_intent_enabled() ) {
```

Comment:
Amended in 2616a6febeb6e3fcbfdc2ce78f1a94bd8b6226c7...Though, having another think about it, in the previous implementation we cleared the legacy UPE intent session regardless of which feature flag is enabled. Perhaps now we should just nuke all session data--legacy UPE and split UPE--to be safer rather than sorry? Let me know which option you'd prefer.

Reply:
I fully agree - it just that the FF check for the deferred UPE wasn't needed. Other than that - this enhancement is more than acceptable and thanks for improvements here!